src/main/java/data/Task.java

```diff
@@ -0,0 +1,135 @@
+package data;
+
+import common.Type;
+
+public class Task {
+    private final String name;
+    private boolean marked;
+    private final Type type;
+    private String time = "";
+
+    public Task(String name, Type type) {
+        this.marked = false;
+        this.name = name;
+        this.type = type;
+    }
+
+    public void mark() {
+        this.marked = true;
+    }
+
+    /**
+     * Set the time of the task.
+     * @param time The time that the task should be done.
+     **/
+    public void setTime(String time) {
+        time = identifyTime(time);
+        if(this.type == Type.EVENT)
+        this.time = "(at: " + time + ")";
+        else if (this.type == Type.DEADLINE) {
+            this.time = "(by: " + time + ")";
+        }
+    }
+
+    /**
+     * Check if the prefix matches this task.
+     * @param prefix the prefix used to check.
+     **/
+    public boolean match(String prefix) {
```

Comment:
Perhaps you can name this method name to has_prefix_match so that is sounds more like boolean